src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+        Order order = init(user, orderRequestDto.address());
+
+        Set<OrderItem> orderItems = orderItemService.loadByCartItems(order,
+                shoppingCart.getCartItems());
+        order.setTotal(countPrice(orderItems));
+        order.setOrderItems(orderItems);
+        cartItemRepository.deleteAllByShoppingCartId(shoppingCart.getId());
+        return orderMapper.toDtoWithoutItems(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderDto> getOrders(Authentication authentication) {
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
+        List<Order> orders = orderRepository.findAllByUserId(user.getId());
+        return orders.stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderWithoutItemsDto updateStatus(Long id, UpdateOrderDto updateOrderDto) {
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by id: " + id));
+        order.setStatus(updateOrderDto.status());
+        return orderMapper.toDtoWithoutItems(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemDto> getItemsByOrderId(Authentication authentication, Long orderId) {
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
+        Order order = isUserOrderCheck(user, orderId);
+        return order.getOrderItems().stream()
+                .map(orderMapper::toOrderItemDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getItemByOrderIdAndItemId(Authentication authentication,
+                                                  Long orderId, Long itemId) {
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
+        Order order = isUserOrderCheck(user, orderId);
+        OrderItem orderItem = orderItemRepository.findByIdAndOrder(itemId, order)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find item with id: "
+                        + itemId + " in order with id: " + orderId));
+        return orderMapper.toOrderItemDto(orderItem);
+    }
+
+    private Order isUserOrderCheck(User user, Long orderId) {
+        return orderRepository.findByIdAndUserId(orderId, user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order with id: "
+                        + orderId + " in user history"));
+    }
+
+    private BigDecimal countPrice(Set<OrderItem> orderItems) {
+        BigDecimal sum = BigDecimal.ZERO;
+        for (OrderItem orderItem : orderItems) {
+            sum = sum.add(orderItem.getPrice());
+        }
+        return sum;
+    }
```

Comment:
did you try to use Stream? https://stackoverflow.com/questions/22635945/adding-up-bigdecimals-using-streams